### PR TITLE
Yelp

### PR DESCRIPTION
**Pull request by bfl3tch**
What I did:
- Created yelp service/facade/new controller to display results of gym search

Testing:
100% sad and happy path

### CODE REVIEW

File: app/poros/gym_from_search.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+class GymFromSearch
+  attr_reader :id, :yelp_gym_id, :name, :address, :phone
+
+  def initialize(info)
+    @id = nil
+    @name = info[:name]
+    @yelp_gym_id = info[:id]
+    @address = info[:location][:display_address].join(", ")
```
Comment (carolinectan) [inline app/poros/gym_from_search.rb:8]: good use of .join

File: app/serializers/found_gym_serializer.rb [hunk 2]
```diff
@@ -0,0 +1,5 @@
+class FoundGymSerializer
```
Comment (carolinectan) [inline app/serializers/found_gym_serializer.rb:1]: possible refactor could be to added to the GymSerializer as a custom method to return all of these attributes

File: spec/poros/gym_from_search_spec.rb [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+                        "state": "ME",
+                        "display_address": [
+                            "2 Livewell Dr",
+                            "Kennebunk, ME 04043"
+                        ]
+                    },
+                    "display_phone": "(207) 467-3800",
+                    "distance": "3621.482123613702"
+                  }
+    end
```
Comment (carolinectan) [inline spec/poros/gym_from_search_spec.rb:29]: Add empty line

File: spec/requests/api/v1/gym_search/index_spec.rb [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+
+        expect(response.status).to eq(200)
+        expect(json).not_to be_empty
+        expect(json_data.size).to eq(20)
+        expect(json_data.first[:attributes][:yelp_gym_id]).to be_a(String)
+      end
+    end
+
+    context 'when a valid location is provided with less than 20 available gyms' do
+      it 'returns 20 nearby gyms sorted by distance', :aggregate_failures, :vcr do
```
Comment (carolinectan) [inline spec/requests/api/v1/gym_search/index_spec.rb:30]: nice use of `, :vcr` for shorter syntax

File: spec/requests/api/v1/gym_search/index_spec.rb [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+        get "/api/v1/gym_search?zip_code=89405"
+        JSON.parse(response.body, symbolize_names: true)
+
+        expect(response.status).to eq(200)
+        expect(json).not_to be_empty
+        expect(json_data.size).to eq(0)
+      end
+    end
+
+    context 'when an invalid location is provided' do
```
Comment (carolinectan) [inline spec/requests/api/v1/gym_search/index_spec.rb:52]: good work testing sad paths

File: spec/services/yelp_service_spec.rb [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+
+      context 'less than 20 gyms nearby but more than 0' do
+        it 'retrieves all the closest gyms on yelp within 40k meters of location', :vcr do
+          location = "04572"
+          results = YelpService.gyms_near_user(location)
+          expect(results[:businesses].count).to eq(16)
+        end
+      end
+
+      context '0 gyms nearby' do
```
Comment (carolinectan) [inline spec/services/yelp_service_spec.rb:30]: good job testing sad paths

**Review by carolinectan — APPROVED** [review]
Good work testing for sad paths and returning different values for different situations for gym searches.

**Review by ealwafai — APPROVED** [review]

**Pull request merged**